core/src/main/scalajvm/scalapb/zio_grpc/server/ZServerCallHandler.scala

```diff
@@ -95,6 +96,51 @@ object ZServerCallHandler {
     streamingInput(
       runtime,
       (req, requestContext, call) =>
-        impl(req).provideEnvironment(ZEnvironment(requestContext)).foreach(call.sendMessage)
+        loop(call, requestContext, impl(req).provideEnvironment(ZEnvironment(requestContext)))
     )
+
+  private def loop[Res](
```

Comment:
Can you add a comment to describe what this function does? What would be a more descriptive name for it other than `loop`?

I'd like to have a unit test for this as a `private[scalapb]` function that cover all branches. It could that a fake implementation of `ZServerCall` if it helps. Doesn't need to be an integration test (no need for sleeps, etc). This will make it easier to bring it changes in ZIO's semantics in the future.

Reply:
Done.